exporters/failure/collector_github.py

```diff
@@ -0,0 +1,189 @@
+        super().__init__(message)
+
+
+class GithubFailureCollector(AbstractFailureCollector):
+    """
+    Github implementation of a FailureCollector
+    """
+
+    REQUIRED_CONFIG = ["TOKEN"]
+    _defaultapi = "api.github.com"
+
+    def __init__(self, apikey, projects, user=None, server=None):
+        super().__init__(server, user, apikey)
+        if projects:
+            self.projects = projects.split(",")
+        else:
+            logging.warning("No projects defined for github issues")
+            self.projects = None
+        try:
+            self.user = self._get_github_user()
+        except Exception:
+            logging.warning("github username not found")
+            self.user = "unknown"
+        self.server = self._defaultapi
+        self.tracker_provider = TRACKER_PROVIDER
+
+    def _get_github_user(self):
+        # login and get username
+        # set the username / server to env for exporter consistency
+        url = "https://{}/user".format(self._defaultapi)
+        headers = {"Authorization": "token {}".format(self.apikey)}
+        resp = self._make_request(headers, url)
+        return resp["login"]
+
+    def _make_request(self, headers, url):
+        this_response = {}
+        s = requests.Session()
+        s.headers.update(headers)
+        # disable .netrc
+        s.trust_env = False
+        resp = s.get(url, headers=headers)
+        if resp.status_code == 200:
+            logging.debug(resp.text)
+            this_response = resp.json()
+        # custom error message
+        elif resp.status_code == 401:
+            raise GithubAuthenticationError()
+        # log the issue w/o exit
+        else:
+            try:
+                resp.raise_for_status()
+            except requests.exceptions.HTTPError as e:
+                logging.error(e)
+        # return empty json if there is an error
+        return this_response
+
+    def get_issues(self):
+        all_issues = []
+        if not self.projects:
+            logging.debug("No github projects are defined")
+        else:
+            for proj in self.projects:
+                logging.info("Getting issues from: {}".format(proj))
+                # note: this is getting issues for each github project
+                url = "https://{}/repos/{}/issues".format(self._defaultapi, proj)
+                headers = {
+                    "Authorization": "token {}".format(self.apikey),
+                    "Accept": "application/vnd.github.v3+json",
+                }
+
+                # all issues or open issues??
+                issues = self._make_request(headers, url)
+                for i in issues:
+                    all_issues.append(i)
+
+        return all_issues
+
+    def search_issues(self):
+        critical_issues = []
+        all_issues = self.get_issues()
+        if not all_issues:
+            logging.debug("No issues were found")
+        else:
+            for issue in all_issues:
```

Comment:
If you dislike unnecessary nesting like I do, you could change this to an early return:

```python
if not all_issues:
    logging.debug("No issues were found")
    return critical_issues
for issue in all_issues:
```